File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/java/dmn/FnInvocationNamedParameters.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"decision.ftl", "fn invocation named parameters"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "fn invocation named parameters",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class FnInvocationNamedParameters extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "fn invocation named parameters",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
        com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private final FnLibrary fnLibrary;

    public FnInvocationNamedParameters() {
        this(new FnLibrary());
    }

    public FnInvocationNamedParameters(FnLibrary fnLibrary) {
        this.fnLibrary = fnLibrary;
    }

    @java.lang.Override()
    public type.TFnInvocationNamedResult applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply((input_.get("inputA") != null ? number(input_.get("inputA")) : null), (input_.get("inputB") != null ? number(input_.get("inputB")) : null), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'FnInvocationNamedParameters'", e);
            return null;
        }
    }

    public type.TFnInvocationNamedResult apply(java.lang.Number inputA, java.lang.Number inputB, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start decision 'fn invocation named parameters'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long fnInvocationNamedParametersStartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments fnInvocationNamedParametersArguments_ = new com.gs.dmn.runtime.listener.Arguments();
            fnInvocationNamedParametersArguments_.put("inputA", inputA);
            fnInvocationNamedParametersArguments_.put("inputB", inputB);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, fnInvocationNamedParametersArguments_);

            // Evaluate decision 'fn invocation named parameters'
            type.TFnInvocationNamedResult output_ = lambda.apply(inputA, inputB, context_);

            // End decision 'fn invocation named parameters'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, fnInvocationNamedParametersArguments_, output_, (System.currentTimeMillis() - fnInvocationNamedParametersStartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'fn invocation named parameters' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<type.TFnInvocationNamedResult> lambda =
        new com.gs.dmn.runtime.LambdaExpression<type.TFnInvocationNamedResult>() {
            public type.TFnInvocationNamedResult apply(Object... args_) {
                java.lang.Number inputA = 0 < args_.length ? (java.lang.Number) args_[0] : null;
                java.lang.Number inputB = 1 < args_.length ? (java.lang.Number) args_[1] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 2 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[2] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                type.TFnLibrary fnLibrary = FnInvocationNamedParameters.this.fnLibrary.apply(context_);

                java.lang.Number subResult = ((com.gs.dmn.runtime.LambdaExpression<java.lang.Number>)(fnLibrary != null ? fnLibrary.getSubFn() : null)).apply(inputA, inputB, context_);
                java.lang.Number multiplicationResultNamed = ((com.gs.dmn.runtime.LambdaExpression<java.lang.Number>)(fnLibrary != null ? fnLibrary.getMultiplyFn() : null)).apply(inputA, inputB, context_);
                java.lang.Number subResultMixed = ((com.gs.dmn.runtime.LambdaExpression<java.lang.Number>)(fnLibrary != null ? fnLibrary.getSubFn() : null)).apply(inputB, inputA, context_);
                java.lang.Number divisionResultNamed = ((com.gs.dmn.runtime.LambdaExpression<java.lang.Number>)(fnLibrary != null ? fnLibrary.getDivideFn() : null)).apply(inputA, inputB, context_);
                type.TFnInvocationNamedResultImpl fnInvocationNamedParameters = new type.TFnInvocationNamedResultImpl();
                fnInvocationNamedParameters.setSubResult(subResult);
                fnInvocationNamedParameters.setMultiplicationResultNamed(multiplicationResultNamed);
                fnInvocationNamedParameters.setSubResultMixed(subResultMixed);
                fnInvocationNamedParameters.setDivisionResultNamed(divisionResultNamed);
                return fnInvocationNamedParameters;
            }
        };
}

File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/kotlin/dmn/FnInvocationNamedParameters.kt

import java.util.*
import java.util.stream.Collectors

@javax.annotation.Generated(value = ["decision.ftl", "fn invocation named parameters"])
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "fn invocation named parameters",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
class FnInvocationNamedParameters(val fnLibrary : FnLibrary = FnLibrary()) : com.gs.dmn.runtime.JavaTimeDMNBaseDecision() {
    override fun applyMap(input_: MutableMap<String, String>, context_: com.gs.dmn.runtime.ExecutionContext): type.TFnInvocationNamedResult? {
        try {
            return apply(input_.get("inputA")?.let({ number(it) }), input_.get("inputB")?.let({ number(it) }), context_)
        } catch (e: Exception) {
            logError("Cannot apply decision 'FnInvocationNamedParameters'", e)
            return null
        }
    }

    fun apply(inputA: java.lang.Number?, inputB: java.lang.Number?, context_: com.gs.dmn.runtime.ExecutionContext): type.TFnInvocationNamedResult? {
        try {
            // Start decision 'fn invocation named parameters'
            var annotationSet_: com.gs.dmn.runtime.annotation.AnnotationSet = context_.getAnnotations()
            var eventListener_: com.gs.dmn.runtime.listener.EventListener = context_.getEventListener()
            var externalExecutor_: com.gs.dmn.runtime.external.ExternalFunctionExecutor = context_.getExternalFunctionExecutor()
            var cache_: com.gs.dmn.runtime.cache.Cache = context_.getCache()
            val fnInvocationNamedParametersStartTime_ = System.currentTimeMillis()
            val fnInvocationNamedParametersArguments_ = com.gs.dmn.runtime.listener.Arguments()
            fnInvocationNamedParametersArguments_.put("inputA", inputA)
            fnInvocationNamedParametersArguments_.put("inputB", inputB)
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, fnInvocationNamedParametersArguments_)

            // Evaluate decision 'fn invocation named parameters'
            val output_: type.TFnInvocationNamedResult? = evaluate(inputA, inputB, context_)

            // End decision 'fn invocation named parameters'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, fnInvocationNamedParametersArguments_, output_, (System.currentTimeMillis() - fnInvocationNamedParametersStartTime_))

            return output_
        } catch (e: Exception) {
            logError("Exception caught in 'fn invocation named parameters' evaluation", e)
            return null
        }
    }

    private inline fun evaluate(inputA: java.lang.Number?, inputB: java.lang.Number?, context_: com.gs.dmn.runtime.ExecutionContext): type.TFnInvocationNamedResult? {
        var annotationSet_: com.gs.dmn.runtime.annotation.AnnotationSet = context_.getAnnotations()
        var eventListener_: com.gs.dmn.runtime.listener.EventListener = context_.getEventListener()
        var externalExecutor_: com.gs.dmn.runtime.external.ExternalFunctionExecutor = context_.getExternalFunctionExecutor()
        var cache_: com.gs.dmn.runtime.cache.Cache = context_.getCache()
        // Apply child decisions
        val fnLibrary: type.TFnLibrary? = this@FnInvocationNamedParameters.fnLibrary.apply(context_)

        val subResult: java.lang.Number? = fnLibrary?.let({ it.subFn as com.gs.dmn.runtime.LambdaExpression<java.lang.Number?>? })?.apply(inputA, inputB, context_) as java.lang.Number?
        val multiplicationResultNamed: java.lang.Number? = fnLibrary?.let({ it.multiplyFn as com.gs.dmn.runtime.LambdaExpression<java.lang.Number?>? })?.apply(inputA, inputB, context_) as java.lang.Number?
        val subResultMixed: java.lang.Number? = fnLibrary?.let({ it.subFn as com.gs.dmn.runtime.LambdaExpression<java.lang.Number?>? })?.apply(inputB, inputA, context_) as java.lang.Number?
        val divisionResultNamed: java.lang.Number? = fnLibrary?.let({ it.divideFn as com.gs.dmn.runtime.LambdaExpression<java.lang.Number?>? })?.apply(inputA, inputB, context_) as java.lang.Number?
        val fnInvocationNamedParameters: type.TFnInvocationNamedResultImpl? = type.TFnInvocationNamedResultImpl() as type.TFnInvocationNamedResultImpl?
        fnInvocationNamedParameters?.subResult = subResult
        fnInvocationNamedParameters?.multiplicationResultNamed = multiplicationResultNamed
        fnInvocationNamedParameters?.subResultMixed = subResultMixed
        fnInvocationNamedParameters?.divisionResultNamed = divisionResultNamed
        return fnInvocationNamedParameters
    }

    companion object {
        val DRG_ELEMENT_METADATA : com.gs.dmn.runtime.listener.DRGElement = com.gs.dmn.runtime.listener.DRGElement(
            "",
            "fn invocation named parameters",
            "",
            com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
            com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
            com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
            -1
        )
    }
}
